pal03377/Athena-New
# Improve module requests details

## Motivation

Currently it is hard to see what is going on in the module requests so some details right there in the playground would be nice to have, as mentioned in #37. Also as preparation for the upcoming (human-)evaluation mode where such detail views are needed.

## Description

- Add details to the module requests (and also to some responses)
   - Exercise details for text and programming exercises
   - Submission details for text and programming exercises 
      - ProgrammingExerciseSubmissionDetail is just a very basic file tree of the submission repository (will follow up in the future)
- Add send all feedback for selected submission option
- Some minor styling changes

Closes #37

## Testing Steps

- `npm install` to update dependencies
- Run playground and some module
- Go through the requests and check that I did not mess up with state management in the request input 
- Verify that everything makes sense and still works as before just with added context :)

## Screenshots

### Updated playground where I did each request for text exercises:
![localhost_3000_playground](https://github.com/ls1intum/Athena/assets/5898705/1818e376-6d57-45c6-9dc1-1202c7d9c102)


### Current exercise detail and submissions detail for programming exercise
![localhost_3000_playground (1)](https://github.com/ls1intum/Athena/assets/5898705/a9e69d15-4422-4e4e-8b4a-5196fbea077d)
